### DISCUSSION TITLE
klipper

### DISCUSSION THREAD

**Discussion by kevinbarrie** [smartin015/continuousprint#174] [Q&A] [answered]
Absolutely love this plugin works amazing.  The problem that we are encountering now is that we would like to run klipper, when we set it up and I try to use it, klipper/octoprint wants the bed homed before running the script, we of course cannot have the printer home as there is a print sitting in the middle of the bed.  Tried the homing override in klipper config and added a general start code to octoprint, but cannot figure out how to make this work seamlessly like without klipper.  We are using this on 10 ender 3 s1's.

Is there a fix for this?

**Comment by smartin015** [comment, marked as answer]
I'm not a klipper user so I can only offer guesses, but it seems like klipper has very strong opinions about homing before doing any kind of movement. Per [here](https://github.com/Klipper3d/klipper/blob/master/docs/FAQ.md#why-cant-i-move-the-stepper-before-homing-the-printer) homing is required to move after steppers are disabled (due to M18 or M84 command). Have you tried checking for / removing these from your .gcode file and cleanup scripts?

Having never used homing override either... you could try setting it to a no-op command, e.g. `G4 P0` to wait 0 seconds - otherwise it may treat "empty list of commands" as "homing override not set".

Also relevant - the `GCODE Scripts` settings section of octoprint **does not play well with Continuous Print**. Those are executed on every run of every file, and CPQ treats clearing/finishing events as files in order to correctly interpret things like [at-commands](https://docs.octoprint.org/en/master/features/atcommands.html). Make sure those are all empty if you're having undesirable clearing behavior. 

HTH

**Reply by kevinbarrie** [reply]
Thanks smartin, turning off disable steppers might do the trick never I will give that a try.
